packages/mongodb-explain-compat/lib/index.js

```diff
@@ -72,18 +72,56 @@ function mapStages(queryPlan, sbeExecutionStages) {
   });
 }
 
-module.exports = function(explain) {
-  explain = JSON.parse(JSON.stringify(explain));
-  if ((explain.explainVersion || 0) < 2) {
+function isAggregationExplain(explain) {
+  return !!explain.stages;
+}
+
+/**
+ * https://wiki.corp.mongodb.com/display/QUERY/Explain+Notes
+ */
+function mapAggregationExplain(explain) {
+  // In aggregations for SBE, the cursor based stage is always first.
+  const firstStage = explain.stages[0]; // {$cursor: {queryPlanner: {}, executionStats: {} }}
```

Comment:
What happens to the other stages?

Reply:
so while executing aggregation, the server will get the necessary data from the collection first and then do the required transformations that are in the pipeline. so, it creates a cursor stage that absorbs all the stages responsible for fetching data where we have all the necessary information for explain. And for rest of the stages, we don't have any `queryPlanner` or `executionStats` information.

And if the cursor absorbs all the stages, then the output of explain is same as find operation (also pipeline is run as find)